test/unit/specs/components/Identity/index.spec.js

```diff
@@ -1,121 +0,0 @@
-    let reactorRegisterGetEcid;
-    let optIn;
-    let identity;
-    let eventManager;
-    beforeEach(() => {
-      reactorRegisterGetEcid = jasmine.createSpy();
-      optIn = {
-        isOptedIn: () => true,
-        whenOptedIn: () => Promise.resolve()
-      };
-      eventManager = {};
-      identity = createIdentity({
-        config: createConfig({
-          reactorRegisterGetEcid
-        }),
-        cookieJar: {
-          get(key) {
-            return key === "ECID" ? "ABC" : null;
-          }
-        },
-        optIn,
-        eventManager
-      });
-    });
-
-    describe("getEcid", () => {
-      it("returns ECID when user is opted in", () => {
-        const getEcid = reactorRegisterGetEcid.calls.first().args[0];
-        expect(getEcid()).toBe("ABC");
-      });
-
-      it("returns undefined when user is not opted in", () => {
-        optIn.isOptedIn = () => false;
-        const getEcid = reactorRegisterGetEcid.calls.first().args[0];
-        expect(getEcid()).toBeUndefined();
-      });
-    });
-
-    describe("onResponse", () => {
-      let cookieJar;
-      let response;
-      beforeEach(() => {
-        cookieJar = {
-          get(key) {
-            return key === "ECID" ? "ABC" : null;
-          },
-          set: jasmine.createSpy()
-        };
-        response = {
-          getPayloadsByType: jasmine
-            .createSpy()
-            .and.returnValue([{ id: "ABC" }]),
-          toJSON: jasmine.createSpy()
-        };
-        optIn = {
-          isOptedIn: () => true,
-          whenOptedIn: () => Promise.resolve()
-        };
-        identity = createIdentity({
-          config: createConfig({
-            reactorRegisterGetEcid
-          }),
-          cookieJar,
-          optIn,
-          eventManager
-        });
-      });
-      it("should get called with an object with a property named response in it", () => {
-        identity.lifecycle.onResponse({ response });
-        return flushPromiseChains().then(() => {
-          expect(response.getPayloadsByType).toHaveBeenCalledWith(
-            "identity:persist"
-          );
-          expect(cookieJar.set).toHaveBeenCalledWith(
-            EXPERIENCE_CLOUD_ID,
-            "ABC"
-          );
-          expect(response.getPayloadsByType).toHaveBeenCalledWith(
-            "identity:exchange"
-          );
-        });
-      });
-      it("should not set ECID if the response doesn't have id", () => {
-        response = {
-          getPayloadsByType: jasmine.createSpy().and.returnValue([]),
-          toJSON: jasmine.createSpy()
-        };
-        identity = createIdentity({
-          config: createConfig({
-            reactorRegisterGetEcid
-          }),
-          cookieJar,
-          optIn
-        });
-        identity.lifecycle.onResponse({ response });
-        return flushPromiseChains().then(() => {
-          expect(cookieJar.set).not.toHaveBeenCalled();
-        });
-      });
-    });
-  });
-});
```

Comment:
Identity tests need to get reworked. We need to be testing `createComponent.js` rather than `index.js`